packages/shared/src/lib/kratos.ts

```diff
@@ -0,0 +1,247 @@
+  text: string;
+  type: string;
+  context?: MetaLabelContext | EmptyObjectLiteral;
+}
+
+interface InitializationNodeMeta {
+  label: NodeMetaLabel;
+}
+
+export interface InitializationNode {
+  type: string;
+  group: string;
+  attributes: InitializationNodeAttribute;
+  messages: ErrorMessage[];
+  meta: InitializationNodeMeta | EmptyObjectLiteral;
+}
+
+interface InitializationUI {
+  action: string;
+  method: string;
+  messages?: ErrorMessage[];
+  nodes: InitializationNode[];
+}
+
+export type RegistrationInitializationData = InitializationData;
+export type AuthenticatorLevel = 'aal0' | 'aal1';
+export type KratosMethod = 'link_recovery' | 'password';
+export type AuthenticationType = 'password' | 'oidc';
+
+interface IdentityCredential {
+  config: Record<string, string>;
+  created_at: Date;
+  identifiers: string[];
+  type: AuthenticationType;
+  updated_at: Date;
+  version: number;
+}
+
+type DateString = string;
+
+export interface InitializationData {
+  id: string;
+  issued_at: DateString;
+  expires_at: DateString;
+  created_at: DateString;
+  updated_at: DateString;
+  request_url: string;
+  type: string;
+  refresh?: boolean;
+  ui: InitializationUI;
+  requested_aal: AuthenticatorLevel;
+}
+
+type Method = 'link_recovery' | 'password';
+
+interface AuthMethod {
+  aal: AuthenticatorLevel;
+  completed_at: Date;
+  method: Method;
+}
+
+interface IdentityTraits {
+  username: string;
+  fullname: string;
+  email: string;
+  image: string;
+}
+
+interface RecoveryAddress {
+  created_at: Date;
+  id: string;
+  updated_at: Date;
+  value: string;
+  via: string;
+}
+
+interface VerifyableAddress extends RecoveryAddress {
+  status: string;
+  verified: boolean;
+  verified_at?: Date;
+  id: string;
+  value: string;
+  via: string;
+  created_at: Date;
+  updated_at: Date;
+}
+
+interface Identity {
+  created_at: Date;
+  credentials?: Record<string, IdentityCredential>;
+  id: string;
+  metadata_admin?: unknown;
+  metadata_public?: unknown;
+  recovery_addresses: RecoveryAddress[];
+  schema_id: string;
+  schema_url: string;
+  state: 'active';
+  state_changed_at: Date;
+  traits: IdentityTraits;
+  updated_at: Date;
+  verifiable_addresses: VerifyableAddress[];
+}
+
+export interface LogoutSessionData {
+  logout_token: string;
+  logout_url: string;
+}
+
+export interface AuthSession extends Partial<LogoutSessionData> {
+  active: boolean;
+  authenticated_at: Date;
+  authentication_methods: AuthMethod[];
+  authenticator_assurance_level: AuthenticatorLevel;
+  expires_at: Date;
+  id: string;
+  identity: Identity;
+  issued_at: Date;
+}
+
+export interface SuccessfulRegistrationData {
+  session: AuthSession;
+  identity: Identity;
+}
+
+export enum AuthFlow {
+  Login = '/login',
+  Logout = '/logout',
+  Registration = '/registration',
+  Settings = '/settings',
+  Recovery = '/recovery',
+  Verification = '/verification',
+}
+
+const authUrl = process.env.NEXT_PUBLIC_AUTH_URL || 'http://127.0.0.1:4433';
```

Comment:
We probably want to move this away to a const file

Reply:
Totally agree.